File: backend/src/ZealEducation.Domain/Entities/StudyMaterial.cs
using ZealEducation.Domain.Common;

namespace ZealEducation.Domain.Entities;

public class StudyMaterial : BaseAuditableEntity
{
    public Guid CourseId { get; set; }
    public Guid UploadedByStaffId { get; set; }
    public string Title { get; set; } = default!;
    public string FileName { get; set; } = default!;
    public string FilePath { get; set; } = default!;
    public string FileType { get; set; } = default!;
    public decimal FileSizeMb { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime UploadedAt { get; set; }

    public Course Course { get; set; } = default!;
    public Staff UploadedByStaff { get; set; } = default!;
    public ICollection<MaterialDownloadLog> DownloadLogs { get; set; } = [];
}

File: frontend/zeal-edu/src/app/core/models/study-material.ts
export interface CourseMaterialsQuery {
  page: number;
  pageSize: number;
  search?: string;
  includeInactive?: boolean;
}

export interface StudyMaterialListItem {
  materialId: string;
  courseId: string;
  title: string;
  fileName: string;
  fileType: string;
  fileSizeMb: number;
  isActive: boolean;
  uploadedAt: string;
  uploadedByName: string;
}
